packages/api/lambdas/ems-ingest-report.js

```diff
@@ -1,7 +1,300 @@
 'use strict';
 
+const fs = require('fs');
 const moment = require('moment');
-const { generateAndSubmitReports, submitReports } = require('../lib/ems');
+const os = require('os');
+const path = require('path');
+const { aws, log } = require('@cumulus/common');
+const {
+  buildReportFileName, determineReportKey, getExpiredS3Objects, reportToFileType, submitReports
+} = require('../lib/ems');
+const { deconstructCollectionId } = require('../es/indexer');
+const { Search, defaultIndexAlias } = require('../es/search');
+
+/**
+ * This module provides functionalities to generate EMS reports.
+ * The following environment variables are used:
+ * process.env.ES_SCROLL_SIZE: default to defaultESScrollSize
+ * process.env.ES_INDEX: set for testing purpose, default to defaultIndexAlias
+ * process.env.ems_provider: default to 'cumulus', the provider used for sending reports to EMS
+ * process.env.ems_submitReport: default to 'false', indicates if the reports will be sent to EMS
+ * process.env.ems_host: EMS host
+ * process.env.ems_port: EMS host port
+ * process.env.ems_path: EMS host directory path for reports
+ * process.env.ems_username: the username used for sending reports to EMS
+ * process.env.ems_privateKey: default to 'ems.private.pem', the private key file used for sending
+ *   reports to EMS. privateKey filename in s3://system_bucket/stackName/crypto
+ * process.env.ems_dataSource: the data source of EMS reports
+ * process.env.ems_retentionInDays: the retention in days for reports and s3 server access logs
+ * process.env.stackName: it's used as part of the report filename
+ * process.env.system_bucket: the bucket to store the generated reports and s3 server access logs
+ */
+
+const defaultESScrollSize = 1000;
+
+const bucketsPrefixes = () => ({
+  reportsBucket: process.env.system_bucket,
+  reportsPrefix: `${process.env.stackName}/ems/`,
+  reportsSentPrefix: `${process.env.stackName}/ems/sent/`
+});
+exports.bucketsPrefixes = bucketsPrefixes;
+
+/**
```

Comment:
Most of the functions in this file are moved here from lib/ems.js, except the cleanup()